account_invoice_payment_ref_mandatory/views/account_move_views.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_account_move_form_inherit" model="ir.ui.view">
+        <field name="name">account.move.form (Ref Mandatory)</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='payment_reference']" position="attributes">
+                <attribute name="required">1</attribute>
+            </xpath>
```

Comment:
Payment reference is currently invisible for certain move types, see:

https://github.com/odoo/odoo/blob/14.0/addons/account/views/account_move_views.xml#L662-L663

Then, with this code, when the move is a journal entry, required and invisible should become incompatible (we should fill a field that is invisible).

Anyway, we'll only set required this field for Vendor operations (`in_invoice` and `in_refund`), we'll extended it later if needed.

Adding view inheritance please check if current `attrs` values (for `invisible` and `readonly`) are kept, or if we should need to copy/paste them in this view.